apps/studio/src/features/editing-experience/components/Preview.tsx

```diff
@@ -13,22 +12,32 @@ export interface PreviewProps {
     content: IsomerPageSchema['content']
   }
 }
+
 export default function Preview({ schema }: PreviewProps) {
   const renderSchema = schema!
+  const [{ theme, isGovernment, sitemap, name }] =
+    trpc.site.getConfig.useSuspenseQuery({ id: 1 })
+  const [{ content: footer }] = trpc.site.getFooter.useSuspenseQuery({
+    id: 1,
+  })
+  const [{ content: navbar }] = trpc.site.getNavbar.useSuspenseQuery({
+    id: 1,
+  })
 
   return (
     <RenderEngine
       site={{
-        siteName: 'Min of ZYX',
+        siteName: name,
+        // TODO: fixup all the typing errors
         // @ts-expect-error blah
+        // TODO: dynamically generate sitemap
         siteMap: { title: 'Home', permalink: '/', children: [] },
-        theme: 'isomer-next',
+        theme,
         logoUrl: 'https://www.isomer.gov.sg/images/isomer-logo.svg',
-        isGovernment: true,
+        isGovernment,
         environment: 'production',
         lastUpdated: '3 Apr 2024',
-        navBarItems: navBar,
-        // @ts-expect-error blah
+        navBarItems: navbar.items,
         footerItems: footer,
```

Comment:
@dcshzj note the difference in calling `navbar/footer` - i think might make more sense to update exported type